packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
+
+Seller signs the first input and output with the sighash type `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` so that any other user can modify the second input and output or add arbitrary inputs and outputs as longs as the input-output pair signed by Seller have the same index. As the input-output pair is signed, Seller is guaranteed that if a transaction containing the pair is included in the blockchain then Seller will get paid. 
 
-#### Sighash Types
+Buyer on the other hand wants to obtain the NFT in the first input so Buyer is incentivized to build the transaction according to the protocol. If the Buyer misbehaves, the worst thing that can happen is that Buyer destroys the NFT and pays the Seller.
 
-The idea is to create a transaction with two inputs and two outputs. The inputs spend the smart object `a` to be sold as well as another smart object `b` that is used for the payment. Like in the case of a swap the smart contract exchanges the owners of the smart objects. However unlike the swap smart contract it returns `[b, a]`. This has the effect that the first output of the transaction will represent `b` after the call and the second will represent `a`.
+### Smart Contracts
 
-Seller then builds the swap transaction and signs it with the [sighash type](https://developer.bitcoin.org/devguide/transactions.html#signature-hash-types) `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` and the input index `0`. This means that Sellers signature remains valid even when arbitrary inputs and outputs are added to the transaction. In particular, another user could change or remove the Bitcoin Computer specific meta data that is located in output 3.
+The first challenge is to build a "crossover" transaction where we have to control the order of inputs and outputs. This is possible with the Bitcoin Computer because the order of inputs is determined by the order of objects in the environment and the order of outputs is determined by the order of objects in the value returned from the expression.
 
-Essentially Seller is saying: you can do with this inputs-output pair whatever you want as long as you spend the output indicated in the first input and you keep the locking script and the amount of the first output the same. Seller can confidentially publish such a partially signed transaction as he wants to sell the nft spent by the fist inputs for the amount indicated in the first output and both are signed.
+Seller calls `encode` with an environment `{ n: ..., p: ... }`, indicating that the first inputs will spend `n` and the second inputs will spend `p`. Seller will use the expression `${Sale} Sale.exec(n, p)`. As the `exec` function of the `Sale` contract returns an array `[p, n]` the first output wil represent `p` and the second output will represent `n`. This is exactly the "crossover" transaction described above.
```

Comment:
Use input in singular: first **input** will spend `n` and the second **input** ...